src/charm.py

```diff
@@ -483,6 +550,31 @@ def _tracing_endpoint_info_grpc(self) -> str:
     def _log_level(self) -> str:
         return self.config["log_level"]
 
+    @property
+    def _internal_url(self) -> str:
+        """Return workload's internal URL. Used for ingress."""
+        return f"http://{socket.getfqdn()}:{ADMIN_UI_PORT}"
```

Comment:
Just a question: I am a bit confused by what `socket.getfqdn` returns in a k8s env. Does it return an FQDN for a Pod or an svc?

If it returns a Pod's FQDN, what happens to the OAuth client config passed to Hydra when scaling up the admin service?

Reply:
It returns the service.

This is not correct in most cases, the redirect_url that we use when registering the client must be accessible by your browser. It won't work unless you expose the service. This is how we do it in other charms, perhaps I should just remove it and block the charm until it has an ingress?

Reply:
ic~ probably better to block and require integrating an ingress so that it won't pass wrong data to Hydra if the ingress is not integrated yet.